src/modules/thirdParty/sagas.ts

```diff
@@ -110,4 +162,106 @@ export function* thirdPartySaga(builder: BuilderAPI) {
+    }
+    return collectionId
+  }
+
+  function* publishChangesToThirdPartyItems(thirdParty: ThirdParty, items: Item[]) {
+    const collectionId = getCollectionId(items)
+    const { signature, signedMessage, salt } = yield call(getPublishItemsSignature, thirdParty.id, items.length)
+
+    const { items: newItems, itemCurations: newItemCurations }: { items: Item[]; itemCurations: ItemCuration[] } = yield call(
+      [builder, 'publishTPCollection'],
+      collectionId,
+      items.map(i => i.id),
+      signedMessage,
+      signature,
+      items.length, // qty
+      salt
+    )
+    return { newItems, newItemCurations }
+  }
+
+  function* handlePublishThirdPartyItemRequest(action: PublishThirdPartyItemsRequestAction) {
+    const { thirdParty, items } = action.payload
+    try {
+      const collectionId = getCollectionId(items)
+      const { newItems, newItemCurations }: { newItems: Item[]; newItemCurations: ItemCuration[] } = yield call(
+        publishChangesToThirdPartyItems,
+        thirdParty,
+        items
+      )
+
+      yield put(publishThirdPartyItemsSuccess(collectionId, newItems, newItemCurations))
+      yield put(closeModal('PublishThirdPartyCollectionModal'))
+    } catch (error) {
+      yield put(publishThirdPartyItemsFailure(error.message))
+    }
+  }
+
+  function* pushChangesToThirdPartyItems(items: Item[]) {
+    const collectionId = getCollectionId(items)
+
+    const itemCurations: ItemCuration[] = yield select(getItemCurations, collectionId!)
+    const effects: CallEffect<ItemCuration>[] = items.map(item => {
+      const curation = itemCurations.find(itemCuration => itemCuration.itemId === item.id)
+      if (curation?.status === CurationStatus.PENDING) {
+        return call([builder, 'updateItemCurationStatus'], item.id, CurationStatus.PENDING)
+      }
+      return call([builder, 'pushItemCuration'], item.id) // FOR CURATIONS REJECTED/APPROVED
+    })
+
+    const newItemsCurations: ItemCuration[] = yield all(effects)
+    return newItemsCurations
+  }
+
+  function* handlePushChangesThirdPartyItemRequest(action: PushChangesThirdPartyItemsRequestAction) {
+    const { items } = action.payload
+    try {
+      const collectionId = getCollectionId(items)
+      const newItemsCurations: ItemCuration[] = yield call(pushChangesToThirdPartyItems, items)
+      yield put(pushChangesThirdPartyItemsSuccess(collectionId, newItemsCurations))
+      yield put(closeModal('PublishThirdPartyCollectionModal'))
+    } catch (error) {
+      yield put(pushChangesThirdPartyItemsFailure(error.message))
+    }
+  }
+
+  function* handlePublishAndPushChangesThirdPartyItemRequest(action: PublishAndPushChangesThirdPartyItemsRequestAction) {
+    const { thirdParty, itemsToPublish, itemsWithChanges } = action.payload
+    const collectionId = getCollectionId(itemsToPublish)
+    // We need to execute this two in secuence, because the push changes will create a new curation if there was one approved.
```

Comment:
```suggestion
    // We need to execute this two methods in sequence, because the push changes will create a new curation if there was one approved.
```

Reply:
We should be using `these` instead of `this` as there are more than one method.